src/goto-symex/symex_goto.cpp

```diff
@@ -329,7 +329,11 @@ void goto_symext::merge_goto(const goto_statet &goto_state, statet &state)
   // do SSA phi functions
   phi_function(goto_state, state);
 
-  merge_value_sets(goto_state, state);
+  // merge value sets
+  if(state.guard.is_false() && !goto_state.guard.is_false())
+    state.value_set = std::move(goto_state.value_set);
+  else if(!goto_state.guard.is_false())
+    state.value_set.make_union(goto_state.value_set);
```

Comment:
It would also make sense for `make_union` to not take a const ref, and use move instead. But that's maybe for another pull request.

Reply:
Making this impactful is a bigger task, because the internal representation of a value set is reference counted. I'll leave this for another time.